module/VuFind/src/VuFind/Db/Mapping/ClassMetadataFactory.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * Extended Doctrine class metadata factory.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Db
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Db\Mapping;
+
+use Doctrine\Persistence\Mapping\ClassMetadata;
+
+/**
+ * Extended Doctrine class metadata factory.
+ *
+ * @category VuFind
+ * @package  Db
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class ClassMetadataFactory extends \Doctrine\ORM\Mapping\ClassMetadataFactory implements ClassMetadataMappingsInterface
+{
+    /**
+     * Alias mappings.
+     *
+     * @var array
+     */
+    protected array $aliases = [];
+
+    /**
+     * Add an alias.
+     *
+     * @param string $alias  Name to be mapped.
+     * @param string $target Target name.
+     *
+     * @return void
+     */
+    public function addAlias(string $alias, string $target): void
+    {
+        $this->aliases[$alias] = $target;
+    }
+
+    /**
+     * Set all aliases.
+     *
+     * @param array $aliases Aliases with names to map as keys and targets as values.
+     *
+     * @return void
+     */
+    public function setAliases(array $aliases): void
+    {
+        $this->aliases = $aliases;
+    }
+
+    /**
+     * Gets the class metadata descriptor for a class.
+     *
+     * @param string $className The name of the class.
+     *
+     * @return ClassMetadata
+     *
+     * @throws ReflectionException
+     * @throws MappingException
```

Comment:
I think these exceptions need to be imported with `use` statements or changed to FQCNs if we want these annotations to be accurate.